source/common/filter/config_discovery_impl.h

```diff
@@ -92,15 +92,6 @@ class DynamicFilterConfigProviderImpl : public DynamicFilterConfigProviderImplBa
     }
   }
 
-  // DynamicFilterConfigProviderImplBase
-  void validateMessage(const std::string& config_name, const Protobuf::Message& message,
-                       const std::string& factory_name) const override {
-    auto* factory = Registry::FactoryRegistry<Factory>::getFactory(factory_name);
```

Comment:
Is `Factory` type parameter still used? I think the hope was to make generic code work for all factories but I see that stat_prefix is odd for HTTP.

Reply:
Yes, the Factory type parameter here is not used any more, neither in class FilterConfigProviderManagerImpl.
Remove it from both places.

BTW, could you please elaborate what's odd for stat_prefix?

Reply:
The instantiation callback takes stat_prefix for HTTP in createFilterFactoryFromProto for historical reasons. So this particular callback cannot be templated across all factories.